G_channels.md

```diff
@@ -249,6 +249,24 @@ socket = assign(socket, :user, msg["user"])
 
 Sockets store assigned values as a map in `socket.assigns`.
 
+#### Failure Semantics
+
+Servers will restart & clients will fail - understanding how Phoenix responds to these events and the guarantees it offers will  help you design robust systems. 
+
+- Subscriptions
+ 
+On the server, subscriptions are held in an ETS table (which resides in memory). If a server restarts or crashes, the subscriptions will need to be recreated for clients to receive updaes. Fortunately, this functionality is built into the Phoenix javascript client. The clients will attempt to reconnect to the server. Once they do, they'll rejoin the channels they were subscribed to using exponential back off.
+
+- Channel Crashes
+
+If a channel crashes on the server, the client is notified and the `channel.onError` callback is invoked. The client then rejoins using exponential back off.
```

Comment:
I think we need to use the plural form when referring to clients here.

If a channel crashes on the server, the clients will be notified and the `channel.onError` callback will be invoked. The clients will then try to rejoin using an exponential back off.